docs_src/microservices/security/Ch-Docker-Swarm-SecuringDeviceServices.md

```diff
@@ -0,0 +1,229 @@
+    # cap_add:
+    #   - "IPC_LOCK"
+    tmpfs:
+      - /vault/config
+    entrypoint: ["/vault/init/start_vault.sh"]
+    environment:
+      - VAULT_ADDR=https://edgex-vault:8200
+      - VAULT_CONFIG_DIR=/vault/config
+      - VAULT_UI=true
+      - SKIP_SETCAP=true
+      - |
+        VAULT_LOCAL_CONFIG=
+          listener "tcp" { 
+              address = "edgex-vault:8200" 
+              tls_disable = "0" 
+              cluster_address = "edgex-vault:8201" 
+              tls_min_version = "tls12" 
+              tls_client_ca_file ="/tmp/edgex/secrets/edgex-vault/ca.pem" 
+              tls_cert_file ="/tmp/edgex/secrets/edgex-vault/server.crt" 
+              tls_key_file = "/tmp/edgex/secrets/edgex-vault/server.key" 
+              tls_perfer_server_cipher_suites = "true"
+          } 
+          backend "consul" { 
+              path = "vault/" 
+              address = "edgex-core-consul:8500" 
+              scheme = "http" 
+              redirect_addr = "https://edgex-vault:8200" 
+              cluster_addr = "https://edgex-vault:8201" 
+          } 
+          default_lease_ttl = "168h" 
+          max_lease_ttl = "720h"
+          disable_mlock = "true"
+    volumes:
+      - vault-file:/vault/file:z
+      - vault-logs:/vault/logs:z
+      - vault-init:/vault/init:ro,z
+      - edgex-vault:/tmp/edgex/secrets/edgex-vault:ro,z
+    depends_on:
+      - consul
+      - security-secrets-setup
+    deploy: 
+      placement: 
+        constraints: 
+          - node.hostname != worker-node 
+```
+
+Another change we had to make is to set the restart policy for one-shot initialization containers like kong-migrations and edgex-proxy.  Simply add this section of yaml to the services you'd like to only run once and they wont be restarted unless a failure condition happens.
+
+```yml
+restart_policy:
+  condition: on-failure
+```
+
+The next and final change in the stack yml file is to ensure the EdgeX services are binding to the correct host.  In Geneva we do this by adding a common variable `Service_ServerBindAddr: "0.0.0.0"` to ensure that the service will bind to any host and not be limited to the hostname.  
+
+### Running the docker stack file
+
+With all of these changes in place we are ready to run the stack file.  We included a script to run the stack file and create the volumes needed in the stack file.  This script simply creates the volumes directories and runs the `docker stack deploy ...` command.
+
+```sh
+sudo ./run.sh
+```
+
+Once the stack is up you can run the following command to view the running services: 
+
+```sh
+sudo docker stack services edgex-overlay
+```
+
+### Confirming results
+
+To ensure the device service is running on the worker node you can run the `docker node ps` command. Now check that you see the device service running on the `worker-node` while all of the other services are running on your host.
```

Comment:
`docker stack ps edgex-overlay` would give more meaningful results, showing what is running where.